### PR TITLE
[WEB-2595] Critical depandancy updates

### PR DESCRIPTION
**Pull request by clintonium-119**
[WEB-2595]

---
Related PRs:
https://github.com/tidepool-org/viz/pull/373
https://github.com/tidepool-org/tideline/pull/436
https://github.com/tidepool-org/platform-client/pull/160

---
A couple of notes regarding packages I would have liked to update, but didn't:
* material-ui-popup-state update: ..
    * requires updating to MUI 5, which would be great, but:
        * requires React >= 17
            * Technically, enzyme still works prior to 18, so this could be done
            * [This article](https://medium.com/swlh/react-v17-nothing-new-but-ed1d3844adc4#:~:text=React%2016%20did%20print%20component,the%20regular%20native%20JavaScript%20stacks.) outlines the changes - nothing to troublesome - would slightly pave the way for eventual 18 upgrade, but not deemed worth the effort for the time being
        * Moving from v4 to v5 would be a lot of code shift, that we just don't have time for presently 
* Emotion
    * Can’t go past 11.5 - breaks unit tests that enzyme.find using displayName (i.e. wrapper.find(‘Button[iconLabel="mylabel"]’)

[WEB-2595]: https://tidepool.atlassian.net/browse/WEB-2595?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by krystophv — CHANGES_REQUESTED** [review]
Just about everything looks good. I'd like to avoid breaking cross-platform compatibility by introducing a bash script for development, though.

File: artifact.test.sh
[the remark this answers is not in the document]
Comment (clintonium-119) [reply artifact.test.sh:1]: Ah yeah, this slipped into VC accidentally at some point in the past.  Removing

File: package.json [hunk 1]
```diff
@@ -1,194 +1,214 @@
...
+    "start": "NODE_ENV=development NODE_OPTIONS='--max-old-space-size=4096' yarn webpack-dev-server --color --progress --port 3000 --host 0.0.0.0",
     "watchViz": "export NODE_OPTIONS='--max-old-space-size=4096' && cd ${TIDEPOOL_DOCKER_VIZ_DIR:-'../viz'} && npm run start",
-    "cleanViz": "./node_modules/.bin/rimraf ${TIDEPOOL_DOCKER_VIZ_DIR:-'../viz'}/dist/*",
-    "startWithViz": "npm run cleanViz; ./node_modules/concurrently/bin/concurrently.js --kill-others --names viz,blip \"npm run watchViz\" \"while ! test -f ${TIDEPOOL_DOCKER_VIZ_DIR:-'../viz'}'/dist/index.js'; do sleep 1; done; npm run start\"",
-    "startLocal": "if node -e 'require(\"./config/local\").listLinkedPackages()' | grep -qF '@tidepool/viz'; then npm run startWithViz; else npm run start; fi",
+    "cleanViz": "yarn rimraf ${TIDEPOOL_DOCKER_VIZ_DIR:-'../viz'}/dist/*",
+    "startWithViz": "npm run cleanViz; yarn concurrently --kill-others --names viz,blip \"npm run watchViz\" \"while ! test -f ${TIDEPOOL_DOCKER_VIZ_DIR:-'../viz'}'/dist/index.js'; do sleep 1; done; npm run start\"",
+    "startLocal": "./config/startLocal.sh",
```
[the remark this answers is not in the document]
Comment (clintonium-119) [reply package.json:18]: I'll revert this.  I hadn't wanted to do it either, but I had changed it to allow it to still run with `yarn startLocal`.
There are some changes in the latest yarn which cause it to error.

Using `npm run startLocal` still works fine, though, and will work for Windows (well, if it ever worked for Windows - I've never tested it)

File: config/startLocal.sh
[the remark this answers is not in the document]
Comment (clintonium-119) [reply config/startLocal.sh:1]: Reverted

**Review by krystophv — APPROVED** [review]
Updates LGTM 🚀

**Pull request merged**